backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/KeyValueHandler.java

```diff
@@ -50,36 +51,46 @@ public KeyValueHandler(Executor executor, FutureJdbi jdbi, String fieldType, Str
     }
   }
 
-  public InternalFuture<List<KeyValue>> getKeyValues(String entityId) {
+  public InternalFuture<List<KeyValue>> getKeyValues(
+      String entityId, List<String> attrKeys, boolean getAll) {
     return jdbi.withHandle(
-        handle ->
-            handle
-                .createQuery(
-                    "select kv_key as k, kv_value as v, value_type as t from "
-                        + getTableName()
-                        + " where entity_name=:entity_name and field_type=:field_type and "
-                        + entityIdReferenceColumn
-                        + "=:entity_id")
-                .bind("entity_id", entityId)
-                .bind("field_type", fieldType)
-                .bind("entity_name", entityName)
-                .map(
-                    (rs, ctx) -> {
-                      try {
-                        return KeyValue.newBuilder()
-                            .setKey(rs.getString("k"))
-                            .setValue(
-                                (Value.Builder)
-                                    CommonUtils.getProtoObjectFromString(
-                                        rs.getString("v"), Value.newBuilder()))
-                            .setValueTypeValue(rs.getInt("t"))
-                            .build();
-                      } catch (InvalidProtocolBufferException e) {
-                        LOGGER.error("Error generating builder for {}", rs.getString("v"));
-                        throw new ModelDBException(e);
-                      }
-                    })
-                .list());
+        handle -> {
+          var queryString =
+              "select kv_key as k, kv_value as v, value_type as t from "
+                  + getTableName()
+                  + " where entity_name=:entity_name and field_type=:field_type and "
+                  + entityIdReferenceColumn
+                  + "=:entity_id";
+
+          if (!attrKeys.isEmpty() && !getAll) {
+            queryString += " AND kv_key IN (<keys>)";
+          }
+          var query = handle.createQuery(queryString);
+          if (!attrKeys.isEmpty() && !getAll) {
+            query.bindList("keys", attrKeys);
+          }
```

Comment:
while getting attributes via `getAttributes ` forget to considered `getAttributeKeysList`, and `getAll` flag which is added now.